src/richie/apps/courses/lms/__init__.py

```diff
@@ -24,8 +28,14 @@ def select_lms(url):
         if url is None:
             return None
 
+        # First check if it matches a configured LMS
         for lms_configuration in settings.RICHIE_LMS_BACKENDS:
             if re.match(lms_configuration.get("COURSE_REGEX", r".*"), url):
                 return import_string(lms_configuration["BACKEND"])(lms_configuration)
 
+        # Second check if it matches Joanie
+        if is_joanie_enabled():
+            if re.match(settings.JOANIE.get("RESOURCE_REGEX", r"^$"), url):
+                return JoanieBackend(settings.JOANIE)
```

Comment:
Actually, this would not be necessary if Joanie was declared as a normal backend in settings...
Is it a bad idea that Joanie could just be an LMS backend like the others?

Reply:
Let's think about that ?

So to configure Joanie, we could use a `DictValue` which default to None ?
```
RICHIE_LMS_BACKENDS = [
        {
            "BASE_URL": "http://localhost:8073",
            "BACKEND": "richie.apps.courses.lms.edx.EdXLMSBackend",
            "COURSE_REGEX": r"^.*/courses/(?P<course_id>.*)/course/?$",
            "JS_BACKEND": "dummy",
            "JS_COURSE_REGEX": r"^.*/courses/(.*)/course/?$",
        },
        values.DictValue(None, JOANIE_LMS_BACKEND, environ_prefix=None)
    ]
```

Then to know if joanie is enabled we could check if we found within `RICHIE_LMS_BACKENDS` a backend with a `BACKEND` equals to `richie.apps.course.lms.joanieJoanieBackend` ?

Reply:
As discussed this morning, let's keep `JOANIE` settings for now then later we will move those settings into a model.